app/controllers/application_controller.rb

```diff
@@ -60,4 +60,15 @@ def basecamp_auth_check
       @basecamp_agent = current_user.agents.where(type: 'Agents::BasecampAgent').first
     end
   end
+
+  def agent_params
+    return {} unless params[:agent]
+    @agent_params ||= begin
+      options = params[:agent].delete(:options) if params[:agent][:options].present?
```

Comment:
For calls that do not send the agent options (agent disable for example) we can not set it to anything, otherwise the validations will fail. See [this spec](https://github.com/cantino/huginn/blob/master/spec/controllers/agents_controller_spec.rb#L315)